db/migrate/20150525081931_remove_duplicate_tokens.rb

```diff
@@ -0,0 +1,28 @@
+class RemoveDuplicateTokens < ActiveRecord::Migration
+  def up
+    # only managed hosts should have tokens
+    Token.where(:id => Token.joins("left outer join hosts on hosts.id = tokens.host_id").
+                             where("hosts.type != 'Host::Managed'").pluck('tokens.id')).delete_all
+
+    # deleting duplicate tokens
+    hosts_with_duplicate_tokens = Token.having('count(*) > 1').group(:host_id).select('host_id')
+    existing_tokens = hosts_with_duplicate_tokens.map(&:token).compact
```

Comment:
This doesn't look quite right to me (untested right now), as `hosts_with_duplicate_tokens` will be an array of host ID ints, but this line (9) is then calling .token on all those ints - they won't be Host objects AFAICT.

Perhaps:

    hosts_with_duplicate_tokens.each do |host_id|
      Token.where(:host_id => host_id).offset(1).destroy_all
    end